### PR TITLE
Iots 1.0.0 m1

### PR DESCRIPTION
**Pull request by charithag**
Add product-mdm IoTS-1.0.0-M1 branch with IoTServer specification

### CODE REVIEW

File: modules/distribution/email-admin-config.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    </footer>
+    <redirectPath>../admin-mgt/update_verifier_redirector_ajaxprocessor.jsp</redirectPath>
+</configuration>
+<configuration type="askPassword">
+<targetEpr></targetEpr>
+<subject>WSO2 MDM - Change password of the new user account</subject>
+<body>
+    Hi {first-name},
+
+
```
Comment (DilanUA) [inline modules/distribution/email-admin-config.xml:49]: unnecessary new line.

File: modules/distribution/email-admin-config.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    Please change your password for the newly created account: {user-name}. Please click the link
+    below to create the password.
+    http://localhost:9443/InfoRecoverySample/infoRecover/verify?confirmation={confirmation-code}&
+    username={user-name}
+
+
```
Comment (DilanUA) [inline modules/distribution/email-admin-config.xml:55]: unnecessary new line.

File: modules/distribution/src/assembly/bin.xml [hunk 3]
```diff
@@ -35,69 +59,37 @@
...
-
-                <exclude>**/sso-idp-config.xml</exclude>
-                <exclude>**/application-authentication.xml</exclude>
-                <exclude>**/sso/jaggery.conf</exclude>
-                <exclude>**/sso/views/register_new_user.jag</exclude>
+                <exclude>**/lib/httpmime*</exclude>
+                <exclude>**/lib/encoder*</exclude>
+                <exclude>**/plugins/jaxb*</exclude>
+                <exclude>**/security/authenticators.xml</exclude>
+
```
Comment (DilanUA) [inline modules/distribution/src/assembly/bin.xml:78]: unnecessary new line.

File: modules/distribution/src/assembly/bin.xml [hunk 4]
```diff
@@ -243,23 +203,54 @@
...
 
+        <!-- <fileSet>
+             <directory>
+                 src/repository/conf/identity/
+             </directory>
+             <outputDirectory>${pom.artifactId}-${pom.version}/repository/conf/identity/</outputDirectory>
+             <includes>
+                 <include>**/**</include>
+             </includes>
+         </fileSet>-->
```
Comment (DilanUA) [inline modules/distribution/src/assembly/bin.xml:230]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/assembly/bin.xml [hunk 5]
```diff
@@ -319,200 +300,100 @@
...
+            </outputDirectory>
+            <includes>
+                <include>authenticationendpoint.war</include>
+                <include>oauth2.war</include>
+                <include>dynamic-client-web.war</include>
+            </includes>
+        </fileSet>
+        <!--copy jaxrs web app for OAuth feature-->
+
+
```
Comment (DilanUA) [inline modules/distribution/src/assembly/bin.xml:357]: unnecessary new line.

File: modules/distribution/src/assembly/bin.xml [hunk 6]
```diff
@@ -539,73 +420,82 @@
...
+                <include>**/rdbms-config.xml</include>
+                <!--<include>**/hbase-analytics-config.xml</include>-->
+                <include>**/analytics-data-config.xml</include>
+                <include>**/analytics-eventsink-config.xml</include>
             </includes>
         </fileSet>
     </fileSets>
 
+
+
```
Comment (DilanUA) [inline modules/distribution/src/assembly/bin.xml:498]: unnecessary new lines.

File: modules/distribution/src/assembly/bin.xml [hunk 7]
```diff
@@ -685,38 +589,21 @@
...
-        <!--<filtered>true</filtered>-->
-        <!--<fileMode>644</fileMode>-->
-        <!--</file>-->
+        <!--        <file>
+                    <source>../p2-profile-gen/target/wso2carbon-core-${carbon.kernel.version}/repository/conf/event-broker.xml
+                    </source>
+                    <outputDirectory>${pom.artifactId}-${pom.version}/repository/conf/</outputDirectory>
+                    <filtered>true</filtered>
+                    <fileMode>644</fileMode>
+                </file>-->
```
Comment (DilanUA) [inline modules/distribution/src/assembly/bin.xml:605]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/assembly/bin.xml [hunk 8]
```diff
@@ -909,119 +879,98 @@
...
+            <outputDirectory>${pom.artifactId}-${pom.version}/repository/conf/identity</outputDirectory>
+            <filtered>true</filtered>
+        </file-->
+
+        <!-- Copying email-admin-config.xml -->
+        <!--file>
+            <source>../p2-profile-gen/target/wso2carbon-core-${carbon.kernel.version}/repository/conf/email/email-admin-config.xml</source>
+            <outputDirectory>${pom.artifactId}-${pom.version}/repository/conf/email</outputDirectory>
+            <filtered>true</filtered>
+        </file-->
```
Comment (DilanUA) [inline modules/distribution/src/assembly/bin.xml:901]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/assembly/bin.xml [hunk 9]
```diff
@@ -909,119 +879,98 @@
...
+        <!-- Copying MDM jaggery app -->
+        <!--<file>-->
+        <!--<source>../apps/jaggery/mdm/target/mdm.zip</source>-->
+        <!--<outputDirectory>${pom.artifactId}-${pom.version}/repository/deployment/server/jaggeryapps/-->
+        <!--</outputDirectory>-->
+        <!--<filtered>false</filtered>-->
+        <!--<fileMode>644</fileMode>-->
+        <!--</file>-->
+
+        <!-- Copying authenticators.xml -->
```
Comment (DilanUA) [inline modules/distribution/src/assembly/bin.xml:941]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/data-bridge/data-bridge-config.xml [hunk 10]
```diff
@@ -1,72 +1,74 @@
...
+             'metaData':[
+             {'name':'ipAdd','type':'STRING'}
+             ],
+             'payloadData':[
+             {'name':'symbol','type':'STRING'},
+             {'name':'price','type':'DOUBLE'},
+             {'name':'volume','type':'INT'}
+             ]
+            }
+        </streamDefinition>
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/data-bridge/data-bridge-config.xml:71]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/data-bridge/data-bridge-config.xml [hunk 11]
```diff
@@ -1,72 +1,74 @@
...
+             {'name':'ipAdd','type':'STRING'}
+             ],
+             'payloadData':[
+             {'name':'symbol','type':'STRING'},
+             {'name':'price','type':'DOUBLE'},
+             {'name':'volume','type':'INT'},
+             {'name':'max','type':'DOUBLE'},
+             {'name':'min','type':'Double'}
+             ]
+            }
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/data-bridge/data-bridge-config.xml:54]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/datasources/cdm-datasources.xml [hunk 12]
```diff
@@ -24,17 +42,58 @@
...
+            <description>The datasource used for CDM Mobile Device Management</description>
             <jndiConfig>
-                <name>jdbc/WSO2AM_DB</name>
+                <name>jdbc/MobileAndroidDM_DS</name>
             </jndiConfig>
             <definition type="RDBMS">
                 <configuration>
-                    <url>jdbc:h2:repository/database/WSO2AM_DB;DB_CLOSE_ON_EXIT=FALSE</url>
+                    <url>jdbc:h2:repository/database/WSO2MobileAndroid_DB;DB_CLOSE_ON_EXIT=FALSE
+                    </url>
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/datasources/cdm-datasources.xml:96]: unnecessary break of new line

File: modules/distribution/src/repository/conf/datasources/cdm-datasources.xml [hunk 13]
```diff
@@ -46,5 +105,27 @@
...
+                    <driverClassName>org.h2.Driver</driverClassName>
+                    <maxActive>50</maxActive>
+                    <maxWait>60000</maxWait>
+                    <testOnBorrow>true</testOnBorrow>
+                    <validationQuery>SELECT 1</validationQuery>
+                    <validationInterval>30000</validationInterval>
+                </configuration>
+            </definition>
+        </datasource>
+
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/datasources/cdm-datasources.xml:129]: unnecessary new line.

File: modules/distribution/src/repository/conf/datasources/master-datasources.xml [hunk 14]
```diff
@@ -46,46 +64,106 @@
...
+        <!--<password>wso2carbon</password>-->
+        <!--<driverClassName>org.h2.Driver</driverClassName>-->
+        <!--<maxActive>50</maxActive>-->
+        <!--<maxWait>60000</maxWait>-->
+        <!--<testOnBorrow>true</testOnBorrow>-->
+        <!--<validationQuery>SELECT 1</validationQuery>-->
+        <!--<validationInterval>30000</validationInterval>-->
+        <!--</configuration>-->
+        <!--</definition>-->
+        <!--</datasource>-->
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/datasources/master-datasources.xml:86]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/datasources/master-datasources.xml [hunk 15]
```diff
@@ -46,46 +64,106 @@
...
+        <!--<configuration>-->
+        <!--<url>jdbc:h2:repository/database/WSO2SOCIAL_CACHE_DB;DB_CLOSE_ON_EXIT=FALSE</url>-->
+        <!--<username>wso2carbon</username>-->
+        <!--<password>wso2carbon</password>-->
+        <!--<driverClassName>org.h2.Driver</driverClassName>-->
+        <!--<maxActive>50</maxActive>-->
+        <!--<maxWait>60000</maxWait>-->
+        <!--</configuration>-->
+        <!--</definition>-->
+        <!--</datasource>-->
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/datasources/master-datasources.xml:104]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/datasources/master-datasources.xml [hunk 16]
```diff
@@ -46,46 +64,106 @@
...
+        <!--<description>The cassandra  datasource used for storing social activities</description>-->
+        <!--<definition type="RDBMS">-->
+        <!--<configuration>-->
+        <!--<url>jdbc:cassandra://localhost:9160/EVENT_KS</url>-->
+        <!--<username>admin@admin.com</username>-->
+        <!--<password>admin</password>-->
+        <!--<driverClassName>org.apache.cassandra.cql.jdbc.CassandraDriver</driverClassName>-->
+        <!--</configuration>-->
+        <!--</definition>-->
+        <!--</datasource>-->
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/datasources/master-datasources.xml:117]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/datasources/master-datasources.xml [hunk 17]
```diff
@@ -46,46 +64,106 @@
...
+        <!--<configuration>-->
+        <!--<url>jdbc:h2:~/test;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=60000</url>-->
+        <!--<username>sa</username>-->
+        <!--<password></password>-->
+        <!--<driverClassName>org.h2.Driver</driverClassName>-->
+        <!--<maxActive>50</maxActive>-->
+        <!--<maxWait>60000</maxWait>-->
+        <!--</configuration>-->
+        <!--</definition>-->
+        <!--</datasource>-->
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/datasources/master-datasources.xml:136]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/identity/service-providers/default.xml [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+                 <IdentityProvider>
+                       <IdentityProviderName>facebook</IdentityProviderName>
+                       <IsEnabled>true</IsEnabled>
+                                     <DefaultAuthenticatorConfig>
+                                             <FederatedAuthenticatorConfig>
+                                                     <Name>FacebookAuthenticator</Name>
+                                                     <IsEnabled>true</IsEnabled>
+                                             </FederatedAuthenticatorConfig>
+                                     </DefaultAuthenticatorConfig>
+                 </IdentityProvider>
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/identity/service-providers/default.xml:53]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/identity/service-providers/sp_cdm.xml [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+                <!-- FederatedIdentityProviders>
+                 <IdentityProvider>
+                       <IdentityProviderName>facebook</IdentityProviderName>
+                       <IsEnabled>true</IsEnabled>
+                                     <DefaultAuthenticatorConfig>
+                                             <FederatedAuthenticatorConfig>
+                                                     <Name>FacebookAuthenticator</Name>
+                                                     <IsEnabled>true</IsEnabled>
+                                             </FederatedAuthenticatorConfig>
+                                     </DefaultAuthenticatorConfig>
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/identity/service-providers/sp_cdm.xml:52]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/identity/service-providers/sp_mdm.xml [hunk 20]
```diff
@@ -0,0 +1,67 @@
...
+                 <IdentityProvider>
+                       <IdentityProviderName>facebook</IdentityProviderName>
+                       <IsEnabled>true</IsEnabled>
+                                     <DefaultAuthenticatorConfig>
+                                             <FederatedAuthenticatorConfig>
+                                                     <Name>FacebookAuthenticator</Name>
+                                                     <IsEnabled>true</IsEnabled>
+                                             </FederatedAuthenticatorConfig>
+                                     </DefaultAuthenticatorConfig>
+                 </IdentityProvider>
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/identity/service-providers/sp_mdm.xml:53]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/security/authenticators.xml [hunk 21]
```diff
@@ -22,39 +24,43 @@
...
+            </Parameter>
+            <Parameter name="NameIDPolicyFormat">
+                urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified
+            </Parameter>
+        </Config>
+
+        <!-- If this authenticator should skip any URI from authentication, specify it under "SkipAuthentication"
+        <SkipAuthentication>
+                <UrlContains></UrlContains>
+            </SkipAuthentication> -->
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/security/authenticators.xml:57]: if not in use, better to remove these lines from the file.

File: modules/distribution/src/repository/conf/security/authenticators.xml [hunk 22]
```diff
@@ -22,39 +24,43 @@
...
+
+        <!-- If this authenticator should skip any URI from session validation, specify it under "SkipAuthentication
+            <SkipSessionValidation>
+                <UrlContains></UrlContains>
+            </SkipSessionValidation> -->
```
Comment (DilanUA) [inline modules/distribution/src/repository/conf/security/authenticators.xml:62]: if not in use, better to remove these lines from the file.

File: modules/distribution/email-admin-config.xml (hunk 1 above)
Comment (charithag) [inline modules/distribution/email-admin-config.xml:49]: This is directly copied form https://github.com/wso2/product-emm/blob/master/modules/distribution/email-admin-config.xml

File: modules/distribution/email-admin-config.xml (hunk 2 above)
Comment (charithag) [inline modules/distribution/email-admin-config.xml:55]: This is directly copied form https://github.com/wso2/product-emm/blob/master/modules/distribution/email-admin-config.xml

File: modules/distribution/src/assembly/bin.xml (hunk 3 above)
Comment (charithag) [inline modules/distribution/src/assembly/bin.xml:78]: Used to separate exclusions from plugins

Comment (charithag) [inline modules/distribution/src/assembly/bin.xml:78]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/assembly/bin.xml (hunk 4 above)
Comment (charithag) [inline modules/distribution/src/assembly/bin.xml:230]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/assembly/bin.xml (hunk 5 above)
Comment (charithag) [inline modules/distribution/src/assembly/bin.xml:357]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513

File: modules/distribution/src/assembly/bin.xml (hunk 6 above)
Comment (charithag) [inline modules/distribution/src/assembly/bin.xml:498]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513

File: modules/distribution/src/assembly/bin.xml (hunk 7 above)
Comment (charithag) [inline modules/distribution/src/assembly/bin.xml:605]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/assembly/bin.xml (hunk 8 above)
Comment (charithag) [inline modules/distribution/src/assembly/bin.xml:901]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/assembly/bin.xml (hunk 9 above)
Comment (charithag) [inline modules/distribution/src/assembly/bin.xml:941]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/data-bridge/data-bridge-config.xml (hunk 10 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/data-bridge/data-bridge-config.xml:71]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/data-bridge/data-bridge-config.xml (hunk 11 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/data-bridge/data-bridge-config.xml:54]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/datasources/cdm-datasources.xml (hunk 12 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/datasources/cdm-datasources.xml:96]: Used to separate url from the remaining configs.

File: modules/distribution/src/repository/conf/datasources/cdm-datasources.xml (hunk 13 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/datasources/cdm-datasources.xml:129]: Used to separate datasource config from each others.

File: modules/distribution/src/repository/conf/datasources/master-datasources.xml (hunk 14 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/datasources/master-datasources.xml:86]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/datasources/master-datasources.xml (hunk 15 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/datasources/master-datasources.xml:104]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/datasources/master-datasources.xml (hunk 16 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/datasources/master-datasources.xml:117]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/datasources/master-datasources.xml (hunk 17 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/datasources/master-datasources.xml:136]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/identity/service-providers/default.xml (hunk 18 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/identity/service-providers/default.xml:53]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/identity/service-providers/sp_cdm.xml (hunk 19 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/identity/service-providers/sp_cdm.xml:52]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/identity/service-providers/sp_mdm.xml (hunk 20 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/identity/service-providers/sp_mdm.xml:53]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/security/authenticators.xml (hunk 21 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/security/authenticators.xml:57]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

File: modules/distribution/src/repository/conf/security/authenticators.xml (hunk 22 above)
Comment (charithag) [inline modules/distribution/src/repository/conf/security/authenticators.xml:62]: Already fixed in commit https://github.com/wso2-incubator/product-iot-server/commit/febdc36cba8b94c3ebcaba27947893ebb2865513 as per the PrabathA's comment.

**Pull request merged**